### PR TITLE
build: service helm chart

### LINKED ISSUE #772: RaaS: Create helm chart for renku service
To make service deployable with renku deployments, we need to rework current Dockerfile to work with uwsgi and create k8s resource files.

### PR DESCRIPTION
**Pull request by rokroskar**
# Description

This PR adds a helm chart for the repository service. Name is still TBD, open to suggestions. 

It's based on #788.  

closes #772

### CODE REVIEW

**Comment by rokroskar** [comment]
> One question - do we want to persist cache states to survive minor outages (like e.g. deployments)?

Yes I considered adding a PV for the cache. I think this can be an improvement added once we have cache cleanup implemented because the service needs to become aware of the cache size. 

> Also, send this PR to my PR?

Thought about that, but I think it makes more sense to keep this separate and rebase once the main PR is merged. Otherwise the other PR becomes impossibly large.

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+version: '3'
+
+services:
+  redis:
+    image: redis:5.0.3-alpine
+    ports:
+      - "6379:6379"
```
Comment (Panaetius) [inline docker-compose.yml]:
```suggestion
      - "6379"
```
We don't need to expose redis to outside the swarm I think.

File: renku/core/management/clone.py [hunk 2]
```diff
@@ -33,23 +34,46 @@ def clone(
...
+
     try:
         repo = Repo.clone_from(
             url, path, recursive=recursive, depth=depth, progress=progress
         )
     except GitCommandError as e:
-        raise errors.GitError(
-            'Cannot clone remote Renku project: {}'.format(url)
-        ) from e
+        if not raise_git_except:
```
Comment (Panaetius) [inline renku/core/management/clone.py]: if `raise_git_except` is `False`, it raises a git exception? Shouldn't it be the other way around?

File: MANIFEST.in [hunk 3]
```diff
@@ -39,6 +41,7 @@ include babel.ini
 include brew.py
 include pytest.ini
 include snap/snapcraft.yaml
+recursive-include renku *.json
```
[the remark this answers is not in the document]
Comment (rokroskar) [reply MANIFEST.in]: Ah this is from the other pr that this one is based on - maybe move the comment there? Apologies for the confusion...

File: Makefile [hunk 4]
```diff
@@ -69,3 +69,6 @@ brew-commit-bottle: *.bottle.json
 
 brew-release:
 	open "https://github.com/SwissDataScienceCenter/renku-python/releases/new?tag=v$(shell brew info --json=v1 renku | jq -r '.[0].versions.stable')"
+
+service-container:
+	docker build -f Dockerfile.svc -t renku-svc .
```
[the remark this answers is not in the document]
Comment (jsam) [reply Makefile]: It's tagged with git hash.

File: renku/core/management/clone.py (hunk 2 above)
Comment (jsam) [reply renku/core/management/clone.py]: Maybe I could improve the wording here. The flag denotes if we should raise the vanilla git exception or raise renku wrapped one. 

Let me try to explain, so one is vanilla git exception and the other is renku wrapped git exception. The wrapped one doesn't really fit the service cause the message attribute is not set within the `__caused__` so I cannot access the explanation part for the reason of the exception (to return it as a response to denote what went wrong). So we need both or find a way to propagate the reason. I can open a ticket for this and look into more detail.

File: renku/service/config.py [hunk 5]
```diff
@@ -55,3 +55,5 @@
     TAR_ARCHIVE_CONTENT_TYPE,
     ZIP_ARCHIVE_CONTENT_TYPE,
 ]
+
+SERVICE_PREFIX = os.getenv('CORE_SERVICE_PREFIX', '/')
```
Comment (jsam) [inline renku/service/config.py]: Can we also add this to the `.env-example`?

File: renku/service/entrypoint.py [hunk 6]
```diff
@@ -36,6 +37,8 @@
     add_file_to_dataset_view, create_dataset_view, dataset_blueprint, \
     list_dataset_files_view, list_datasets_view
 
+logging.basicConfig(level=logging.DEBUG)
```
Comment (ableuler) [inline renku/service/entrypoint.py]: Maybe use "WARNING" as default log level?

File: helm-chart/renku-core/templates/tests/test-connection.yaml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+  labels:
+{{ include "renku-core.labels" . | indent 4 }}
+  annotations:
+    "helm.sh/hook": test-success
+spec:
+  containers:
+    - name: wget
+      image: busybox
+      command: ['wget']
+      args:  ['{{ include "renku-core.fullname" . }}:{{ .Values.service.port }}']
```
Comment (ableuler) [inline helm-chart/renku-core/templates/tests/test-connection.yaml]: Does this actually work? Maybe query the `/health` endpoint to test the connection?

Comment (rokroskar) [reply helm-chart/renku-core/templates/tests/test-connection.yaml]: oh I think this was automatically generated  by `helm` - I will remove it

**Comment by rokroskar** [comment]
follow-up issue: https://github.com/SwissDataScienceCenter/renku-python/issues/890

**Pull request merged**